onnx/defs/tensor_proto_util.cc

```diff
@@ -34,12 +34,29 @@ namespace ONNX_NAMESPACE {
 #define DEFINE_PARSE_DATA(type, typed_data_fetch)                              \
   template <>                                                                  \
   const std::vector<type> ParseData(const TensorProto* tensor_proto) {         \
-    std::vector<type> res;                                                     \
     if (!tensor_proto->has_data_type() ||                                      \
       tensor_proto->data_type() == TensorProto_DataType_UNDEFINED) {           \
       fail_shape_inference("The type of Tensor: ", tensor_proto->name(),       \
         " is undefined so it cannot be parsed.");                              \
-    } else if (tensor_proto->has_data_location() &&                            \
+    } else if (std::is_same<type, int64_t>::value  &&                          \
```

Comment:
you should simplify this check
You can change the macro to take the enum type field something like this:

DEFINE_PARSE_DATA(int32_t, int32_data, TensorProto_DataType_INT32)

and then directly compare tensor_proto->data_type() == enum_type

Reply:
Good idea. Updated.